tests/Core/Generators/GeneratorTest.php

```diff
@@ -0,0 +1,240 @@
+                    $pathToDocsInFixture.'Structure'.DIRECTORY_SEPARATOR.'TwoStandardBlocksNoCodeStandard.xml',
+                    $pathToDocsInFixture.'Structure'.DIRECTORY_SEPARATOR.'TwoStandardBlocksOneCodeComparisonStandard.xml',
+                    $pathToDocsInFixture.'Structure'.DIRECTORY_SEPARATOR.'TwoStandardBlocksThreeCodeComparisonsStandard.xml',
+                ],
+            ],
+        ];
+
+    }//end dataConstructor()
+
+
+    /**
+     * Verify that an XML doc which isn't valid documentation yields an Exception to warn devs.
+     *
+     * This should not be hidden via defensive coding!
+     *
+     * @return void
+     */
+    public function testGeneratingInvalidDocsResultsInException()
+    {
+        // Set up the ruleset.
+        $standard = __DIR__.'/NoValidDocsTest.xml';
+        $config   = new ConfigDouble(["--standard=$standard"]);
+        $ruleset  = new Ruleset($config);
+
+        if (PHP_VERSION_ID >= 80000) {
+            $exception = 'TypeError';
+            $message   = 'processSniff(): Argument #1 ($doc) must be of type DOMNode, null given';
+        } else if (PHP_VERSION_ID >= 70000) {
+            $exception = 'TypeError';
+            $message   = 'processSniff() must be an instance of DOMNode, null given';
+        } else {
+            $exception = 'PHPUnit_Framework_Error';
+            $message   = 'processSniff() must be an instance of DOMNode, null given';
+        }
+
+        if (method_exists($this, 'expectExceptionMessage') === true) {
+            // PHPUnit 5.2.0+.
+            $this->expectException($exception);
+            $this->expectExceptionMessage($message);
+        } else {
+            // Ancient PHPUnit.
+            $this->setExpectedException($exception, $message);
+        }
+
+        $generator = new MockGenerator($ruleset);
+        $generator->generate();
+
+    }//end testGeneratingInvalidDocsResultsInException()
+
+
+    /**
+     * Verify the wiring for the generate() function.
+     *
+     * @param string $standard The standard to use for the test.
+     * @param string $expected The expected function output.
+     *
+     * @dataProvider dataGeneratingDocs
+     *
+     * @return void
+     */
+    public function testGeneratingDocs($standard, $expected)
+    {
+        // Set up the ruleset.
+        $config  = new ConfigDouble(["--standard=$standard"]);
+        $ruleset = new Ruleset($config);
+
+        // Make the test OS independent.
+        $expected = str_replace("\n", PHP_EOL, $expected);
+        $this->expectOutputString($expected);
+
+        $generator = new MockGenerator($ruleset);
+        $generator->generate();
+
+    }//end testGeneratingDocs()
+
+
+    /**
+     * Data provider.
+     *
+     * @return array<string, array<string, string>>
+     */
+    public static function dataGeneratingDocs()
+    {
+        $multidocExpected  = "No Content\n";
```

Comment:
Good point. I think it's because I started with defining `expected` as a multi-line string, which would have the `\n` line ending as that's how the test file is saved, so needed the new line conversion, and then when I iterated on the test, I just never changed it.

Fixed now.